### PR TITLE
Fix cyclic dependency of apiserver test for OpenAPI test

### LINKED ISSUE #41748: genericapiserver needs a non-cyclical test for openapi
The existing openapi test for genericapiserver had a cycle in it.  We need a simple test for enablement that doesn't introduce a cycle.  https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_test.go#L91
 
@mbohlool @sttts

### PR DESCRIPTION
**Pull request by mbohlool**
Fixes #41748

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/genericapiserver_test.go [hunk 1]
```diff
@@ -80,6 +80,12 @@ func init() {
 	examplev1.AddToScheme(scheme)
 }
 
+func testGetOpenAPIDefinitions(_ kubeopenapi.ReferenceCallback) map[string]kubeopenapi.OpenAPIDefinition {
```
Comment (lavalamp) [inline staging/src/k8s.io/apiserver/pkg/server/genericapiserver_test.go]: Is this function actually called?

Comment (lavalamp) [reply staging/src/k8s.io/apiserver/pkg/server/genericapiserver_test.go]: nvm, I saw the call site the instant I sent the comment.

**Comment by lavalamp** [comment]
/lgtm
/approve

Tests pass and it removes dead code, so it seems like a good idea-- is it really still WIP?

**Comment by mbohlool** [comment]
code change is just rebase. applied lgtm manually.

**Pull request merged**